packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4742,3 +4744,46 @@ export class Gitxnas extends Gtxnas {
 		super.execute(stack);
 	}
 }
+
+/**
+ * Takes the last value from stack and if base64encoded, decodes it acording to the
+ * encoding e and pushes it back to the stack, otherwise throws an error
+ */
+export class Base64Decode extends Op {
+	readonly line: number;
+	readonly encoding: number;
+
+	/**
+	 * Asserts 1 argument is passed.
+	 * @param args Expected arguments: [e], where e = {0, 1}.
+	 * @param line line number in TEAL file
+	 */
+	constructor(args: string[], line: number) {
+		super();
+		this.line = line;
+		assertLen(args.length, 1, line);
+		this.encoding = Number(args[0]);
+		if (!(this.encoding === 0 || this.encoding === 1)) {
+			throw new RuntimeError(RUNTIME_ERRORS.TEAL.UNKNOWN_ENCODING, {
+				encoding: this.encoding,
+				line: this.line,
+			});
+		}
+	}
+
+	execute(stack: TEALStack): void {
+		this.assertMinStackLen(stack, 1, this.line);
+		const last = this.assertBytes(stack.pop(), this.line);
+		const enc = new TextDecoder("utf-8");
+		const decoded = enc.decode(last);
+		if (this.encoding === 0) {
+			// UrlEncoding
+			assertBase64Url(convertToString(last), this.line);
+			stack.push(new Uint8Array(Buffer.from(decoded.toString(), "base64url")));
+		} else {
+			// e === 1 StdEncoding
+			assertBase64(convertToString(last), this.line);
```

Comment:
you're running the same code in both `if else`. Move it outside

Reply:
Actually they are different. One is `base64` while the other is `base64Url`